### PR TITLE
Added Vector ds

### PR DESCRIPTION
**Pull request by gargVader**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Vector is one of the most commonly used data structure from the C++ STL library. But many people are not aware how it works. The implementation has all the functions like push_back(), pop_back(), front() etc.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1447

### CODE REVIEW

File: C-Plus-Plus/ds/Vector.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	T operator[](const int i) {
+		return arr[i];
+	}
+
+	// Clears the vector
+	void clear() {
+		cs = 0;
+	}
+
+
```
Comment (iamrajiv) [inline C-Plus-Plus/ds/Vector.cpp]: @gargVader Remove unnecessary line break.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@gargVader Add sample input-output in code comments.

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — CHANGES_REQUESTED** [review]
There are functions like ```empty() , pop_back()``` that aren't being demonstrated in the driver code. 

**Suggestions :**
- You may make it a menu driven program to allow users to explore various aspects of vector data structure on their own.
- Prompt user to enter the values in the vector.
- In ```T operator[](const int i)``` what if i entered is an index out of bound ?

**Comment by gargVader** [comment]
- [x] Menu added for users
- [x] Prompted users to enter values in the vector
- [x] If i entered is an index out of bounds then, we cannot predict the output.

File: C-Plus-Plus/ds/Vector.cpp [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	cout << "Enter <1> To push elements \n"
+	     "<2> To pop elements \n"
+	     "<3> To get front element \n"
+	     "<4> To get back element\n"
+	     "<5> To check if vector is empty \n"
+	     "<6> To get size of vector \n"
+	     "<7> To get capacity of vector \n"
+	     "<8> To get iterator to start of vector \n"
+	     "<9> To get end of vector \n"
```
Comment (atarax665) [inline C-Plus-Plus/ds/Vector.cpp]:
```suggestion
	     "<9> To get iterator to the end of vector \n"
```

File: C-Plus-Plus/ds/Vector.cpp [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+void printVector(Vector<int> v) {
+	for (int i = 0; i < v.size(); i++) {
+		cout << v[i] << " ";
+	} cout << endl;
+}
+
+int main() {
+
+	cout << "Enter <1> To push elements \n"
+	     "<2> To pop elements \n"
```
Comment (atarax665) [inline C-Plus-Plus/ds/Vector.cpp]:
```suggestion
	     "<2> To pop (last) element \n"
```

File: C-Plus-Plus/ds/Vector.cpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+		case 1: cout << "Enter elements to push until -1: ";
+			int x;
+			cin >> x;
+			while (x != -1) {
+				v.push_back(x);
+				cin >> x;
+			} break;
+
+		case 2: v.pop_back(); break;
```
Comment (atarax665) [inline C-Plus-Plus/ds/Vector.cpp]: Do check if a vector has any element in it before performing push_back().If the vector is empty return a message saying ```The vector is empty.```.  As calling pop_back() on an empty vector results in undefined behavior.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/Vector.cpp [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+		case 1: cout << "Enter elements to push until -1: ";
+			int x;
+			cin >> x;
+			while (x != -1) {
+				v.push_back(x);
+				cin >> x;
+			} break;
+
+		case 2: if(!v.empty()) cout << v.pop_back() << endl;
```
Comment (atarax665) [inline C-Plus-Plus/ds/Vector.cpp]:
```suggestion
		case 2: if(!v.empty()) v.pop_back(); 
```
v.pop_back() function just pops out the last element in the vector, it is a void function so it won't return any value to be printed using cout.

**Review by atarax665 — CHANGES_REQUESTED** [review]
The overall code looks fine now, thank you for your patient contribution. Just a minor change and then it would be good to merge. :+1:

File: C-Plus-Plus/ds/Vector.cpp (hunk 5 above)
Comment (gargVader) [reply C-Plus-Plus/ds/Vector.cpp]: Yeah. It is so embarrassing

**Review by atarax665 — APPROVED** [review]
LGTM :+1:

**Pull request merged**